lua/script.lua

```diff
@@ -3,106 +3,69 @@
 
 local Script = {}
 
---- reset script environment;
--- ie redirect draw, key, enc functions, stop timers, clear engine, etc
-Script.clear = function()
-  print("# script clear")
-  -- reset cleanup script
-  cleanup = norns.none
-  -- reset oled redraw
-  redraw = norns.blank
-  -- redirect inputs to nowhere
-  key = norns.none
-  enc = norns.none
-  -- clear, redirect, and reset grid
-  if g then 
-    g:all(0)
-    g:refresh()
-    g.key = norns.none
-  end
-  g = nil
-  -- reset gridkey callback
-  gridkey = norns.none
-  -- reset midi callbacks
-  midi.add = norns.none
-  midi.remove = norns.none
-  -- stop all timers
-  metro.free_all()
-  -- stop all polls and clear callbacks
-  poll.clear_all()
-  -- clear engine
-  engine.name = nil
-  -- clear init
-  init = norns.none
-  -- clear last run
-  norns.state.script = ''
-  norns.state.name = 'none'
-  -- clear params
-  params:clear()
-end
+Script.file = ''
+Script.name = 'none' 
+Script.errorstate = false
```

Comment:
a thought.

if `errormsg` were set to `nil` when not in an error state, you could drop this flag and just check for `nil` instead.

not sure that's better though| 🤔